### PR TITLE
perf(amd): tune GFX1250 DSA and KDA prefill kernels

### PR DESCRIPTION
**Pull request by Yu-Zhewen**
## DSA prefill changes

- Radix top-k block size: `4096` → `2048`.
- Prefill logits block size: `32` → `128`.
- Prefill logits warps: `4` → `8`.
- Synchronize the radix histogram after every subpass.
- Bound radix output writes by the number of qualifying entries.
- Use 64-bit-addressed stores for dense-attention outputs exceeding the 2 GiB buffer-offset limit.

| Batch | Sequence length | Before | After | Difference |
|---:|---:|---:|---:|---:|
| 1 | 1K | 193.29 µs | 124.31 µs | -35.7% |
| 1 | 4K | 2.357 ms | 989.49 µs | -58.0% |
| 1 | 8K | 8.172 ms | 2.595 ms | -68.2% |
| 8 | 1K | 6.366 ms | 1.330 ms | -79.1% |
| 8 | 4K | 96.452 ms | 17.578 ms | -81.8% |
| 8 | 8K | 367.206 ms | 60.489 ms | -83.5% |
| 32 | 1K | 91.050 ms | 13.605 ms | -85.1% |
| 32 | 4K | 1,429.879 ms | 205.198 ms | -85.6% |
| 32 | 8K | 6,037.544 ms | 789.627 ms | -86.9% |

## KDA prefill changes

- Make the cumulative gate scan wave-local, reducing LDS and VGPR usage while doubling residency.
- Asynchronously TDM-load Q/K into LDS and overlap the loads with gate computation.
- State-scan output block: `16` → `8`.
- State-scan `waves_per_eu`: `1` → `2`.
- Output-kernel `waves_per_eu`: `1` → `4`.

| Batch | Sequence length | Before | After | Difference |
|---:|---:|---:|---:|---:|
| 1 | 1K | 144.95 µs | 124.61 µs | -14.0% |
| 1 | 4K | 312.80 µs | 272.84 µs | -12.8% |
| 1 | 8K | 565.76 µs | 486.75 µs | -14.0% |
| 8 | 1K | 383.91 µs | 336.80 µs | -12.3% |
| 8 | 4K | 1.323 ms | 1.090 ms | -17.6% |
| 8 | 8K | 2.555 ms | 2.117 ms | -17.1% |
| 32 | 1K | 1.333 ms | 1.095 ms | -17.8% |
| 32 | 4K | 5.063 ms | 4.150 ms | -18.0% |
| 32 | 8K | 10.089 ms | 8.251 ms | -18.2% |

### CODE REVIEW

File: tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/attention/kda/prefill.py [hunk 1]
```diff
@@ -335,13 +337,30 @@ def _preprocess_intra_fwd_kernel(
...
-    k_norm = gl.rsqrt(gl.sum(k_value * k_value, axis=1) + 1e-6)
-    normalized_q = (q_value * q_norm[:, None]).to(gl.bfloat16)
-    normalized_k = (k_value * k_norm[:, None]).to(gl.bfloat16)
-    gl.store(kn + offsets, normalized_k, mask=mask)
+    shared_qk: gl.constexpr = gl.PaddedSharedLayout.with_identity_for(
+        [[K, 8]], [BT, K], [1, 0]
+    )
+    shared_bg: gl.constexpr = gl.SwizzledSharedLayout(4, 1, 8, order=[1, 0])
+    q_smem = gl.allocate_shared_memory(gl.bfloat16, [BT, K], shared_qk)
+    k_smem = gl.allocate_shared_memory(gl.bfloat16, [BT, K], shared_qk)
```
[the remark this answers is not in the document]
Comment (Yu-Zhewen) [reply tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/attention/kda/prefill.py]: Fixed in 8852e86c. The TDM LDS buffers and normalized WMMA operands now use the Q/K input element types, so FP16 payloads remain FP16 instead of being interpreted as BF16. Verified both BF16 and FP16 against the recurrent reference (2 passed); per request, no test-file change was committed.
